src/build/mod.rs

```diff
@@ -182,7 +182,7 @@ impl Build {
         }
     }
 
-    fn as_pending(self) -> PendingBuild {
+    fn assert_pending(self) -> PendingBuild {
```

Comment:
Nit: Personally, it'd find it surprising that an `assert_*` method returns any value. Maybe something like `as_pending_checked` would fit better, if we're changing the name?

Reply:
This was a clippy convention fix, `as_` should use `&self` (https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#wrong_self_convention). Ie it's surprising to consume `self` with `as_` methods. A simple fix would be to rename to `into_pending`. 

But since it panics if the state is unexpected, to me this is also surprising behaviour. I think maybe `try_into_pending` is the proper one.